runtime/compiler/optimizer/SequentialStoreSimplifier.cpp

```diff
@@ -1061,6 +1061,14 @@ bool isValidSeqLoadByteConversion(TR::Compilation* comp, bool trace, TR::Node* c
        * Under 64bit, bloadi's child must be an aladd node. Some of the aladd nodes go away after the transformation.
        * So, if the refCount is not 1, bail out here.
        */
+
+      // If accessing the first element with OffHeap enabled, then it can be an
+      // aloadi dataAddrPointer and there would be no second child offset.
+      if (firstChild->isDataAddrPointer())
+         {
+         return true
+         }
+
```

Comment:
For the sequential load simplification part of the opt, this check guards the rest of the opt. From here on, it assumes the `aladd` node exists and won't check again. If you bypass it like this, other things can break.

These two definitely won't work anymore:
getALoadReferenceForSeqLoad:
https://github.com/eclipse-openj9/openj9/blob/73af0c72ea6d61f98bc7b07ff2d6c02128a71d61/runtime/compiler/optimizer/SequentialStoreSimplifier.cpp#L1362-L1402
getBaseOffsetForSeqLoad:
https://github.com/eclipse-openj9/openj9/blob/73af0c72ea6d61f98bc7b07ff2d6c02128a71d61/runtime/compiler/optimizer/SequentialStoreSimplifier.cpp#L1231-L1289

There may be others.